src/shogun/machine/GLM.h

```diff
@@ -0,0 +1,291 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Both the documentation and the code is heavily inspired by pyGLMnet:
+ * https://github.com/glm-tools/pyglmnet/
+ *
+ * Author: Tej Sukhatme
+ */
+
+#include <shogun/lib/config.h>
+
+#include <shogun/distributions/Distribution.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/IterativeMachine.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/machine/Machine.h>
+#include <shogun/mathematics/RandomMixin.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/optimization/ConstLearningRate.h>
+#include <shogun/optimization/ElasticNetPenalty.h>
+#include <shogun/optimization/FirstOrderCostFunction.h>
+#include <shogun/optimization/GradientDescendUpdater.h>
+
+namespace shogun
+{
+	enum GLM_DISTRIBUTION
+	{
+		POISSON
+	};
+
+	class DotFeatures;
+	class Features;
+	class RegressionLabels;
+	class GLMCostFunction;
+
+	/** @brief Class for estimating regularized generalized linear models (GLM).
+	 *   The regularized GLM minimizes the penalized negative log likelihood:
+	 *
+	 *  This uses Elastic-net penalty which defaults to the ridge penalty when
+	 *  alpha = 0 and defaults to the lasso penalty when alpha = 1.
+	 *
+	 * */
+	class GLM : public RandomMixin<IterativeMachine<LinearMachine>>
+	{
+	public:
+		// friend class GLMCostFunction;
```

Comment:
If you want to remove code, just remove it. Don’t just comment it out. Please don’t commit this kind of changes.

Reply:
Comment removed